tests/test_RecursiveFetaureElimination.py

```diff
@@ -1,6 +1,98 @@
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.datasets import make_friedman1
+from sklearn.linear_model import LinearRegression
+
 from feature_selection import recursive_feature_elimination
 
 
-def test_1():
-    result = recursive_feature_elimination(1, 2, 3, 4, 5)
-    assert result == []
+def scorer(X, y):
+    """
+    Sample custom scorer that fits a model and returns
+    an appropriate score for the feature selection problem.
+
+    Parameters
+    ----------
+    X : array-like of shape (n_samples, n_features)
+        Test samples
+
+    y : array-like of shape (n_samples,) or (n_samples, n_outputs)
+        True values for X, used for training
+
+    Returns
+    -------
+    Index of feature column with the lowest weight.
+    """
+    model = LinearRegression()
+    model.fit(X, y)
+    return X.columns[model.coef_.argmin()]
+
+
+def test_recursive_feature_elimination():
+    """
+    This test creates a dataset that has 5 features that are are used to compute `y`.
+    The remaining 5 features are independent of `y`.
+    This test should select the 5 feature columns used to compute `y`.
+    """
+    X, y = make_friedman1(n_samples=200, n_features=10, random_state=10)
+    features = recursive_feature_elimination(scorer, X, y, n_features_to_select=5)
+    assert features == [0, 1, 3, 4, 9]
```

Comment:
I think for friedman, significant features are [0, 1, 2, 3, 4], although 2 is the weakest of them all. I recommend double checking this test a few times, because 9 could be a random pick and mess up future tests. A more reliable test would be to do this:

```suggestion
    features = recursive_feature_elimination(scorer, X, y, n_features_to_select=4)
    assert features == [0, 1, 3, 4]
```

Reply:
> Out of the n_features features, only 5 are actually used to compute y. The remaining features are independent of y.

For RFE, I'm not considering any ranking. There are 5 features used to compute `y` and I verified my results with scikit-learn's `LinearRegression` estimator. The important thing is that RFE picked the correct 5.

I can change it to `4` but I don't think it matters.

Reply:
That's the problem though. Both RFE and your model picked an **incorrect feature.** It just so happens that at random_state = 10, feature 9 is identified. If you remove the random state in friedman1, you'll notice that the fifth chosen feature varies widely. 

Take a look here:
https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_friedman1.html

Notice how feature 9 is random. If your feature selector chose feature 9, technically it was incorrect, so I dislike that your test validates to what sklearn RFE incorrectly did, instead of actually identifying the correct features. This could cause problems if I say passed in a different scorer than linear regression (say RandomForestRegressor), which will pick feautres [0, 1, 2, 3, 4] for random_state = 10 (I tested this in Jupyter Lab) and your test will incorrectly fail even though that answer is correct.

Reply:
Ok, I see what you mean now. I didn't realize that it's always the first 5 features that are relevant.